File: server/src/protocol/SensorTagSamplesBuilder.cpp
#include "SensorTagSamplesBuilder.h"
#include "SensorTagSamplesCommand.h"

using namespace std;
using namespace communication;

SensorTagSamplesBuilder::SensorTagSamplesBuilder()
{}

SensorTagSamplesBuilder::~SensorTagSamplesBuilder()
{}

unique_ptr<Command> SensorTagSamplesBuilder::create(const vector<uint8_t> &commandInBytes)
{
    auto command = make_unique<SensorTagSamplesCommand>();

    return move(command);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/SensorTagSamplesBuilder.java
package wpam.mobile_client.protocol;

import java.io.Closeable;
import java.util.ArrayList;

public final class SensorTagSamplesBuilder implements CommandBuilder, Closeable
{
    public SensorTagSamplesBuilder() {
    }
    public final void close() {
    }

    @Override
    public Command create(ArrayList<Integer> commandInBytes)  {
        Command command  = new SensorTagSamplesCommand();
        return command;
    }
}